src/libtriton/engines/solver/bitwuzla/bitwuzlaSolver.cpp

```diff
@@ -226,6 +226,29 @@ namespace triton {
       }
 
 
+      triton::uint512 BitwuzlaSolver::evaluate(const triton::ast::SharedAbstractNode& node) const {
+        if (node == nullptr) {
+          throw triton::exceptions::AstTranslations("BitwuzlaSolver::evaluate(): node cannot be null.");
+        }
+
+        // Perform check-sat on empty solver to enable model generation.
+        auto bzla = bitwuzla_new();
+        bitwuzla_set_option(bzla, BITWUZLA_OPT_PRODUCE_MODELS, 1);
+        if (bitwuzla_check_sat(bzla) != BITWUZLA_SAT) {
+          bitwuzla_delete(bzla);
+          return 0;
```

Comment:
Is it a possible control flow or should we throw an exception here?

Reply:
MMmmh that's a good question because for me it's cannot be unsat as there is no symbolic variable? There is no equation neither? For example we could be able to evaluate `2 + 3 * 4`, so there is no SAT reasoning here.

Reply:
In this case we call `bitwuzla_check_sat()` on empty solver, so its always should be SAT. The concrete AST is translated and sent to solver later. Querying check-sat is necessary, because without it solver won't allow getting values from it.

Reply:
Can you add what you wrote into a comment to keep this understanding when reading code?